oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elasticsearch/query/ElasticsearchIndexStatistics.java

```diff
@@ -16,46 +16,144 @@
  */
 package org.apache.jackrabbit.oak.plugins.index.elasticsearch.query;
 
+import com.google.common.base.Ticker;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.ListenableFutureTask;
 import org.apache.jackrabbit.oak.plugins.index.elasticsearch.ElasticsearchConnection;
+import org.apache.jackrabbit.oak.plugins.index.elasticsearch.ElasticsearchIndexDefinition;
 import org.apache.jackrabbit.oak.plugins.index.search.IndexStatistics;
 import org.elasticsearch.client.RequestOptions;
 import org.elasticsearch.client.core.CountRequest;
 import org.elasticsearch.client.core.CountResponse;
 import org.elasticsearch.index.query.QueryBuilders;
 import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.annotations.TestOnly;
 
 import java.io.IOException;
+import java.util.Objects;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
 
+/**
+ * Cache-based {@code IndexStatistics} implementation providing statistics for Elasticsearch reducing
+ * network operations.
+ * <p>
+ * By default, statistic values expire after 10 minutes but are refreshed in background when accessed after 1 minute.
+ */
 class ElasticsearchIndexStatistics implements IndexStatistics {
+
+    private static final LoadingCache<CountRequestDescriptor, Integer> DEFAULT_STATS_CACHE =
+            setupCache(10000, 10L, 1L, null);
```

Comment:
Thinking out loud - Would it make sense to make this configurable if it might turn out to be useful to tune this parameters specific to customer requirements at some point of time ?

Reply:
done